test/client/query-subscribe.js

```diff
@@ -11,586 +11,579 @@ module.exports = function(options) {
-        query.on('remove', function(docs, index) {
-          expect(util.pluck(docs, 'id')).eql(['fido']);
-          expect(util.pluck(docs, 'data')).eql([undefined]);
-          expect(index).a('number');
-          var results = util.sortById(query.results);
-          expect(util.pluck(results, 'id')).eql(['spot']);
-          expect(util.pluck(results, 'data')).eql([{age: 5}]);
-          done();
-        });
-      });
-    });
+      commonCustomChannelsErrorHandlingTests();
+      commonTests(options);
 
-    it('subscribed query removes document from results before sending delete op to other clients', function(done) {
-      var connection1 = this.backend.connect();
-      var connection2 = this.backend.connect();
-      var matchAllDbQuery = this.matchAllDbQuery;
-      async.parallel([
-        function(cb) {
-          connection1.get('dogs', 'fido').on('error', done).create({age: 3}, cb);
-        },
-        function(cb) {
-          connection1.get('dogs', 'spot').on('error', done).create({age: 5}, cb);
-        }
-      ], function(err) {
-        if (err) return done(err);
-        var query = connection2.createSubscribeQuery('dogs', matchAllDbQuery, null, function(err) {
-          if (err) return done(err);
-          connection1.get('dogs', 'fido').del();
-        });
-        query.on('error', done);
-        var removed = false;
-        connection2.get('dogs', 'fido').on('del', function() {
-          expect(removed).equal(true);
-          done();
+      describe('multiple same channels', function() {
+        beforeEach(function() {
+          this.backend.use('commit', function(context, next) {
+            context.channels.push('channel-1');
+            context.channels.push('channel-2');
+            next();
+          });
+          this.backend.use('query', function(context, next) {
+            context.channels = ['channel-1', 'channel-2'];
+            next();
+          });
         });
-        query.on('remove', function(docs, index) {
-          removed = true;
-          expect(util.pluck(docs, 'id')).eql(['fido']);
-          expect(util.pluck(docs, 'data')).eql([{age: 3}]);
-          expect(index).a('number');
-          var results = util.sortById(query.results);
-          expect(util.pluck(results, 'id')).eql(['spot']);
-          expect(util.pluck(results, 'data')).eql([{age: 5}]);
+
+        it('does not duplicate messages', function(done) {
+          var connection = this.backend.connect();
+          var count = 0;
+          var query = connection.createSubscribeQuery(
+            'dogs',
+            this.matchAllDbQuery,
+            {pollInterval: 0, pollDebounce: 0},
+            function(err) {
+              if (err) return done(err);
+              connection.get('dogs', count.toString()).on('error', done).create({});
+            }
+          );
+          query.on('error', done);
+          query.on('insert', function() {
+            count++;
+            if (count === 3) {
```

Comment:
Alec - let's add a `> 3` condition that calls done with error, so it will catch cases where duplicate causes it to go above 3

Reply:
Changed